### PR TITLE
[3006.x] Draft: 65264 retain existing perms (DEB)

### LINKED ISSUE #65264: [BUG] System packages should not override existing permissions
**Description**

Since adding the `salt` user and running the master as the new user we modify permissions on several directories and files when installing salt (deb and rpm packages). This works great but some users want to run the master as a user other than salt. In those cases; modifying the permissions at package install time cause users to have to re-adjust those permissions every time they upgrade salt.

Ideally the packages can be smart enough to not modify permissions if the target directories already exist and are owned by another user. Perhaps checking the permissions and only modifying them if they are owned by root is enough.

See also: #65243

### PR DESCRIPTION
**Pull request by barneysowood**
### What issues does this PR fix or reference?
Fixes: #65264 

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by max-arnold** [comment]
This PR might be affected by https://github.com/saltstack/salt/pull/66030 (or vice versa, depending on which one will be merged first)

**Comment by barneysowood** [comment]
splitting this - keeping this for Debian packages and moving RPM changes to #66050 as the mechanics of how this will work are quire different

File: pkg/debian/salt-master.preinst [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
 
-    # Reset permissions to fix previous installs
-    find ${SALT_HOME} /etc/salt /var/log/salt /var/cache/salt /var/run/salt \
-        \! \( -path /etc/salt/cloud.deploy.d\* -o -path /var/log/salt/cloud -o -path /opt/saltstack/salt/lib/python${PY_VER}/site-packages/salt/cloud/deploy\* \) -a \
-        \( -user ${SALT_USER} -o -group ${SALT_GROUP} \) -exec chown root:root \{\} \;
+    case "$2" in
+      # if upgrading from 3006.0/3006.1/3006.2 reset permissions
+      # for salt user - issue 64193 and 65264
+
+      3006.0|3006.1|3006.2)
```
Comment (dmurphy18) [inline pkg/debian/salt-master.preinst:8]: these are pretty old versions of Salt , what about later versions eg 3006.7, 3007.0 etc.
Can this be done a different way ? that is not so version specific ?

**Review by dmurphy18 — COMMENTED** [review]
wondering if we can get make it not so version specific, esp. since 3005.x is EOL

**Comment by dmurphy18** [comment]
@barneysowood Given this PR is not getting a lot of attention, wondering if you would be fine with my taking it over

**Comment by dmurphy18** [comment]
Closing this in favor of PR https://github.com/saltstack/salt/pull/66218

**Pull request closed without merging**